pyccel/ast/variable.py

```diff
@@ -823,3 +825,12 @@ def lhs(self):
         The lhs is a
         """
         return self._lhs
+
+    def __eq__(self, other):
+        if isinstance(other, DottedVariable):
+            return self.name == other.name and self.lhs == other.lhs
```

Comment:
If you subclass this class and forget to override `__eq__`, this function will always return `False`. You should check if `self` and `other` have the same type instead, as you did before in the same file.

Reply:
You have this the wrong way round. `isinstance(other, DottedVariable)` returns `True` if `other` is a `DottedVariable` or a sub-class of `DottedVariable`, whereas `type(self) is type(other)` will only return `True` if both types are identical

Reply:
if class X inherits from  DottedVariable and we want that X should never be equal to DottedVariable
we should use ```type(self) is type(other)``` otherwise we can keep it like this

Reply:
Sorry, you're right. Nevertheless, the code is still wrong :wink:

An equality comparison must return `False` if the two types are not identical.

Reply:
My last reply was for @EmilyBourne, but I need to manually refresh this webpage in order to see your replies, and so I am always late... @MenoIy Please use `type(self) is type(other)` as you do in other places.

Reply:
done a592199

Reply:
Thanks